infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -129,7 +146,7 @@ public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
      * @return LogMetadata constructed from all the addresses in the
      * consecutive segments from [startSegment, endSegment]
      */
-    private LogMetadata initializeLogMetadata(long startSegment, long endSegment) {
+    private LogMetadata initializeLogMetadata(long startSegment, long endSegment, long startingAddress) {
```

Comment:
Why do you need the startingAddress here?

Reply:
Because we already read it from the dataStore, we just pass it as a parameter we don't need to read it form the datastore second time

Reply:
If you are passing startingAddress, then you can derive startSegment from it as well. Maybe just have initializeLogMetadata(long startingAddress, long endSegment)